arm-wt-22k/lib_src/eas_mdls.c

```diff
@@ -1358,9 +1371,26 @@ static EAS_RESULT Parse_data (SDLS_SYNTHESIZER_DATA *pDLSData, EAS_I32 pos, EAS_
         }
         else
         {
-            for(i=0; i<count; i++)
+            switch(pWsmp->fmtTag)
             {
-                *p++ = (short)((convBuf[i] ^ 0x80) << 8);
+                case WAVE_FORMAT_ALAW:
+                    for(i=0; i<count; i++)
+                    {
+                        *p++ = alaw2linear(convBuf[i]);
+                    }
+                    break;
+                case WAVE_FORMAT_MULAW:
+                    for(i=0; i<count; i++)
+                    {
+                        *p++ = ulaw2linear(convBuf[i]);
+                    break;
+                }
```

Comment:
Please format correctly:
```suggestion
                    }
                    break;
```

Reply:
Sorry about that